apps/wakunode2/app.nim

```diff
@@ -329,6 +300,71 @@ proc setupWakuApp*(app: var App): AppResult[void] =
 
   ok()
 
+proc getPorts(listenAddrs: seq[MultiAddress]):
+              AppResult[tuple[tcpPort, websocketPort: Option[Port]]] = 
+  
+  var tcpPort, websocketPort = none(Port)
+  
+  for a in listenAddrs:
+    if a.isWsAddress() and websocketPort.isNone():
+      let wsAddress = initTAddress(a).valueOr:
+        return err(error)
+      websocketPort = some(wsAddress.port)
+    elif not a.isWsAddress() and tcpPort.isNone():
```

Comment:
It's a nitpick, but how about

```
if a.WsAddress():
  ...handle wsPort...
else:
  ...handle nonWsPort...
```

Just to skip doing the ws address check twice

Reply:
Another thing - this will obviously take a port from the first listen address it encounters - is that possible an issue? Could it mean we use non-public port instead of a public one, or is that not  a problem anymore with all the previous changes regarding listen addresses?

Reply:
Thank you! Just refactored it to avoid doing the ws address check twice :)

Regarding the ports, the proc takes the tcp and ws(s) ports the switch got bound to - there should be only one of each (this `listenAddrs` seq is part of the switch's struct).
Afterwards we recreate the App's and WakuNode's netConfig as if we had manually configured to bind to those ports.

These lines should not concern public ports: if we configured previously an `extPort`, then NetConfig will be recreated with the same `extPort` as before. And on the other hand, if we use the `ext-multi-addr-only` CLI flag, we would still only (and blindly) announce the multiaddresses  passed by the user.